### PR TITLE
Remove unsupported -s WORKAROUND_IOS_9_RIGHT_SHIFT_BUG=1 option.

### PR DESCRIPTION
**Pull request by juj**
Remove unsupported -s WORKAROUND_IOS_9_RIGHT_SHIFT_BUG=1 option. Fix uses of MIN_SAFARI_VERSION.

Tested with
```a.c
#include <emscripten.h>

int main()
{
	int a = EM_ASM_INT(return 1);
	int b = EM_ASM_INT(return 0);
	int c = a >> b;
	EM_ASM(console.log($0), c);
}
```

that Wasm2JS does produce offending code that might fail on iPhone 4s and older:

```js
function __original_main() {
 var $0 = 0, $1 = 0, $2 = 0;
 $0 = __stack_pointer - 16 | 0;
 __stack_pointer = $0;
 HEAP8[$0 + 15 | 0] = 0;
 $1 = emscripten_asm_const_int(1024, $0 + 15 | 0, 0) | 0;
 HEAP8[$0 + 14 | 0] = 0;
 $2 = emscripten_asm_const_int(1033, $0 + 14 | 0, 0) | 0;
 HEAP8[$0 + 12 | 0] = 105;
 HEAP8[$0 + 13 | 0] = 0;
 HEAP32[$0 >> 2] = $1 >> $2; // (*)
 emscripten_asm_const_int(1042, $0 + 12 | 0, $0 | 0) | 0;
 __stack_pointer = $0 + 16 | 0;
 return 0;
}
```

See https://github.com/emscripten-core/emscripten/pull/7191 for the original issue.

Fortunately our min spec is no longer at iPhone 4s, but higher, so it is not an issue that Wasm2JS does not work for devices so old.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -1387,8 +1387,8 @@ def default_setting(name, new_default):
       shared.Settings.MIN_EDGE_VERSION = 0
       shared.Settings.MIN_CHROME_VERSION = 0
 
-    if shared.Settings.MIN_SAFARI_VERSION <= 9 and shared.Settings.WASM2JS:
-      shared.Settings.WORKAROUND_IOS_9_RIGHT_SHIFT_BUG = 1
+    if shared.Settings.MIN_SAFARI_VERSION <= 90305:
+      exit_with_error('MIN_SAFARI_VERSION <= 90305: Emscripten does not support iPhone 4s, iPad 2, iPad 3, iPad Mini 1, Pod Touch 5 (devices with end-of-life at iOS 9.3.5) and older. (browser has no Wasm support, and Wasm2JS support would not run). Please pass -s MIN_SAFARI_VERSION=100000 or higher.')
```
Comment (kripken) [inline emcc.py:1391]: There is an `if shared.Settings.LEGACY_VM_SUPPORT:` in this file that then sets `MIN_SAFARI_VERSION` to 0. It seems like that would conflict with erroring here, making that flag no longer work.

I think the error is too extreme. How about not erroring, and adding some docs to `settings.js` to see that "0" means 90305 or later? Even if not stated in the others, we surely have other such minimal versions in all browsers.

Comment (juj) [reply emcc.py]: > Even if not stated in the others, we surely have other such minimal versions in all browsers.

Agree. I view this rather that we have just not done the work to identify those minimal versions on other browsers. I think it would be nice to be explicit about those minimal versions in other browsers as well, since that will carry valuable information to users.

However dropped this diagnostic, how does that look now?

**Comment by juj** [comment]
Looks like CI failures that are unrelated to this PR (sse, avx, and code size tests)

File: tests/test_browser.py [hunk 2]
```diff
@@ -2509,7 +2509,7 @@ def test_doublestart_bug(self):
     '': ([],),
     'closure': (['-O2', '-g1', '--closure', '1', '-s', 'HTML5_SUPPORT_DEFERRING_USER_SENSITIVE_REQUESTS=0'],),
     'pthread': (['-s', 'USE_PTHREADS', '-s', 'PROXY_TO_PTHREAD'],),
-    'legacy': (['-s', 'MIN_FIREFOX_VERSION=0', '-s', 'MIN_SAFARI_VERSION=0', '-s', 'MIN_IE_VERSION=0', '-s', 'MIN_EDGE_VERSION=0', '-s', 'MIN_CHROME_VERSION=0'],)
+    'legacy': (['-s', 'MIN_FIREFOX_VERSION=0', '-s', 'MIN_SAFARI_VERSION=90306', '-s', 'MIN_IE_VERSION=0', '-s', 'MIN_EDGE_VERSION=0', '-s', 'MIN_CHROME_VERSION=0'],)
```
Comment (kripken) [inline tests/test_browser.py:2512]: I'm still missing something. Why is this needed - can't it stay at 0 like in all the others?

I see 0 as meaning "as far back as we can". It's ok if that means version X+ and not back til the beginning of history :smile:

Comment (juj) [reply tests/test_browser.py:2512]: This was out of date. Updated back to 0.

**Comment by juj** [comment]
Ping, this could use another look.

**Review by kripken — APPROVED** [review]
